docs/installation/webserver/nginx.md

```diff
@@ -6,106 +6,263 @@ Nginx: Configuring the virtual host
-    # this location block to prevent site visitors having access to the 
-    # various directories that contain executable code.
-    #
-    # NOTE: This approach is *not* recommended for production use.
-    #
-    #location ~ /(?:app|src|tests|vendor)/(.*)$ {
-    #    deny all;
-    #}
-
-    # Block hidden files
-    location ~ /\. {
-        deny all;
-    }
-
-    # Block access to Sqlite database files
-    location ~ /\.(?:db)$ {
-        deny all;
-    }
-
-    # Block access to certain JSON files
-    location ~ /(?:bower|composer|jsdoc|package)\.json$ {
-        deny all;
-    }
-
-    # Block access to Markdown, Twig & YAML files directly
-    location ~* /(.*)\.(?:dist|markdown|md|twig|yaml|yml)$ {
-        deny all;
-    }
-
-    location ~ \.php$ {
-        fastcgi_pass 127.0.0.1:9000;
-        fastcgi_index index.php;
-        include fastcgi_params;
-        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
-        fastcgi_param HTTPS off;
-    }
+    server_name                example.com;
+
+    # Logging
+    access_log                 /var/log/nginx/example.com.access.log;
+    error_log                  /var/log/nginx/example.com.error.log;
+
+    # Site root
+    root                       /var/www/sites/example.com/public;
+    index                      index.php;
+
+    # Bolt specific
+    include                    global/bolt.conf;
+
+    # PHP FPM
+    include                    global/fpm.conf;
+
+    # Let's Encrypt Certbot challenges (optional)
+    #include                    global/acme-challenge.conf;
+
+    # Restrictions
+    include                    global/restrictions.conf;
 }
 ```
 
-**Note:** 502 Bad Gateway Errors
 
-If you're using UNIX sockets instead of TCP ports on your PHP-FPM installation,
-you will need to change the `fastcgi_pass` parameters to match what is set in
-your PHP-FPM configuration's `listen = `, e.g.:
+**NOTE:** You will need to customise/change the values for the `server_name`,
+`access_log`, `error_log` and `root` parameters to match the domain name and
+path locations relevant to your system.
+
+
+### Common
+
+Common, or "global", configuration settings are stored in a directory under
+`/etc/nginx/`. For our examples we use the `/etc/nginx/global/` directory for
+no other reason besides semantics.
+
+This section contains details on the following files:
+
+| File name             | Description |
+| --------------------- | ----------- |
+| `bolt.conf`           | Bolt specific routes
+| `restrictions.conf`   | Files & directories to block access to
+| `php-fpm.conf`        | PHP-FPM configuration
+| `acme-challenge.conf` | (optional) Let's Encrypt Certbot challenges
+
+
+#### `bolt.conf`
 
+The `bolt.conf` file define location maches common to all of your Bolt sites on
+a host.
+
+```nginx
+# Redirect requests for */index.php to the same route minus the "index.php"
+if ($request_uri ~* "^(.*/)index\.php(?>/|)(.*)$") {
```

Comment:
https://www.nginx.com/resources/wiki/start/topics/depth/ifisevil/

`rewrite ^/index.php/(.*) /$1  permanent;` should work, right?

Reply:
OK, no. That article has context and your assertion misses that context. While I'll agree that most the time you don't want this, by the time a request as fallen though to this point you're really splitting hairs. Also if someone has a site that is trafficked to the point where this matters, they are not reading these documentation pages :wink:

Reply:
Yeah, agreed that it won't impact anyone very much, I was mainly thinking of "best practices" and that most of the time when comparing `$request_uri` in a `if` you usually want a location block (or a rewrite if that is the goal anyway) since a location block will match against request_uri anyway, right?